react/AddressForm.tsx

```diff
@@ -1,49 +1,165 @@
+  const summary = display.extended as LineFragment[][]
+  const [ignoredFields, setIgnoredFields] = useState(
+    getSummaryFields(display.compact)
+  )
+
+  const hasWithoutNumberOption = (label: string) => {
+    return label.length >= 6 && label.substr(label.length - 6) == 'Option'
+  }
+
+  const getFieldProps = (field: Field, fragment: LineFragment) => {
+    const { maxLength, autoComplete, required, label, options } = field
+
+    const onChange = (
+      event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+    ) => {
+      setAddress({
+        ...address,
+        [fragment.name]: event.target.value,
+      })
+    }
+    const fieldRequired = {
+      errorMessage: intl.formatMessage(messages.fieldRequired),
+    }
+    const value = address[fragment.name]
+
+    return {
+      label: (
+        <FormattedMessage {...messages[label as keyof (typeof messages)]} />
+      ),
+      value,
+      onChange,
+      ...(options && { options }),
+      ...(maxLength && { maxLength }),
+      ...(autoComplete && { autoComplete }),
+      ...(required && address[fragment.name].length === 0 && fieldRequired),
+    }
+  }
 
   const parseLineFragment = (fragment: LineFragment) => {
-    return address[fragment.name] != null ? (
+    const field = fields[fragment.name as keyof Fields]
+    if (
+      !field ||
+      ignoredFields.has(fragment.name) ||
+      address[fragment.name] == null
+    )
+      return null
+
+    if (hasWithoutNumberOption(field.label))
+      return <NumberOption showCheckbox />
+
+    return (
       <span key={fragment.name} className="w-25 dib mh3">
-        <Input
-          placeholder={fragment.name}
-          label={fragment.name}
-          value={address[fragment.name]}
-          onChange={(event: any) => {
-            const newAddress = {
-              ...address,
-              [fragment.name]: event.target.value,
-            }
-            setAddress(newAddress)
-          }}
-        />
+        {field.options ? (
+          <Dropdown {...getFieldProps(field, fragment)} />
+        ) : (
+          <Input {...getFieldProps(field, fragment)} />
+        )}
       </span>
-    ) : null
+    )
   }
 
   const parseLine = (line: LineFragment[], index: number) => [
     ...line.map(parseLineFragment),
-    <br className={'line' + (index + 1) + '-delimiter'} key={index} />,
+    <br key={index} />,
   ]
 
+  const onEditButtonClick = () => {
+    setIgnoredFields(getSummaryFields(display.minimal))
+    setEditing(true)
+  }
+
+  const displayMode: keyof Display = editing ? 'minimal' : 'compact'
+
   return (
     <div>
+      <PlaceDetails display={displayMode} />
+      {!editing && (
+        <ButtonPlain onClick={onEditButtonClick} title="edit">
+          <FormattedMessage {...messages.edit} />
+        </ButtonPlain>
+      )}
       <div>{summary.map(parseLine)}</div>
     </div>
   )
 }
 
-export default AddressForm
+export default injectIntl(AddressForm)
```

Comment:
I think you can use the [`useIntl`](https://github.com/formatjs/react-intl/blob/master/docs/API.md#useintl-hook) hook from `react-intl`, if you prefer that